front/app/components/admin/FormResults/FormResultsQuestion/SentimentQuestion/utils.tsx

```diff
@@ -1,35 +1,82 @@
+import { colors } from '@citizenlab/cl2-component-library';
+import { lighten } from 'polished';
 import { Multiloc } from 'typings';
 
-import { ResultUngrouped } from 'api/survey_results/types';
+import { ResultGrouped, ResultUngrouped } from 'api/survey_results/types';
 
-export type SentimentAnswers =
-  | {
-      answer: number;
-      count: number;
-      percentage: number;
-      label: string | Multiloc | undefined;
-    }[]
-  | undefined;
+export type SentimentAnswer = {
+  answer: number | null;
+  count: number;
+  percentage: number;
+  label: Multiloc | undefined;
+};
+
+export type SentimentAnswers = SentimentAnswer[] | undefined;
 
-export const parseResult = (result: ResultUngrouped) => {
-  const usersNoAnswer = result.answers?.[5].count;
+// parseResult:
+// Parses survey results and extracts sentiment-related data.
+export const parseResult = (
+  result: ResultUngrouped | ResultGrouped
+): SentimentAnswers => {
+  const usersNoAnswer = result.answers?.[5].count; // Assumes index 5 corresponds to "no answer"
   const totalUsersWhoAnswered = usersNoAnswer
     ? result.totalPickCount - usersNoAnswer
     : result.totalPickCount;
 
   const { multilocs } = result;
 
-  const resultsArray = result.answers?.map((answer) => {
+  return result.answers?.map((answer) => {
+    const resultValue = answer.answer;
+
     return {
-      answer: parseInt(answer.answer?.toString() || '', 10),
+      answer: resultValue && parseInt(resultValue.toString(), 10),
       count: answer.count,
       percentage: Math.round((answer.count / totalUsersWhoAnswered) * 100),
-      label:
-        answer.answer === null
-          ? 'No answer'
-          : multilocs?.answer[answer.answer].title_multiloc,
+      label: resultValue && multilocs?.answer[resultValue].title_multiloc,
     };
   });
+};
+
+// getSentimentGroupColour:
+// Retrieves the corresponding color for a sentiment group (low, neutral, high).
+export const getSentimentGroupColour = (
+  sentimentGroup: string
+): string | undefined =>
+  ({
+    low: colors.red400,
+    neutral: colors.grey400,
+    high: colors.green400,
+  }[sentimentGroup]);
+
+// getSentimentValueColour:
+// Retrieves the corresponding color for a specific sentiment value (1-5).
+export const getSentimentValueColour = (answer: number): string | undefined =>
+  ({
+    1: colors.red400,
+    2: lighten(0.2, colors.red400),
+    3: colors.grey400,
+    4: lighten(0.2, colors.green400),
+    5: colors.green400,
+  }[answer]);
+
+// getPercentageDifference:
+// Calculates the percentage difference between two periods.
+export const getPercentageDifference = (
+  thisPeriodAvg: number,
+  lastPeriodAvg: number
+): number => {
+  return ((thisPeriodAvg - lastPeriodAvg) / lastPeriodAvg) * 100;
+};
```

Comment:
We should probably prevent division by zero here

```suggestion
export const getPercentageDifference = (
  thisPeriodAvg: number,
  lastPeriodAvg: number
): number => {
  if (lastPeriodAvg === 0) {
    return 0;
  }
  return ((thisPeriodAvg - lastPeriodAvg) / lastPeriodAvg) * 100;
};
```